src/activity_item_handling.cpp

```diff
@@ -7,266 +9,383 @@
-    g->u.activity.values.push_back( to_vehicle );
-    add_drop_pairs( selected_worn_items, worn_item_quantities );
-    add_drop_pairs( selected_items, item_quantities );
+    const std::string part_name = veh.part_info( part ).name();
+
+    if( same_type( items ) ) {
+        const item &it = items.front();
+        const int dropcount = items.size() * ( it.count_by_charges() ? it.charges : 1 );
+        add_msg( ngettext( "You put your %1$s in the %2$s's %3$s.",
+                           "You put your %1$s in the %2$s's %3$s.", dropcount ),
+                 it.tname( dropcount ).c_str(), veh.name.c_str(), part_name.c_str() );
+    } else {
+        add_msg( _( "You put several items in the %1$s's %2$s." ), veh.name.c_str(), part_name.c_str() );
+    }
+
+    if( fallen_count > 0 ) {
+        add_msg( m_warning, _( "The trunk is full, so some items fell to the %2$s." ), ter_name.c_str() );
+    }
 }
 
-static tripoint get_item_pointers_from_activity(
-    std::list<item *> &selected_items, std::list<int> &item_quantities,
-    std::list<item *> &selected_worn_items, std::list<int> &worn_item_quantities )
+void stash_on_pet( const std::vector<item> &items, monster &pet )
 {
-    // Drop activity has indices of items in inventory and quantities of same.
-    // Indices and quantities alternate on the list.
-    // First iterate over the indices and quantities, retrieving item references.
-    for( size_t index = 0; index < g->u.activity.values.size(); index += 2 ) {
-        const int position = g->u.activity.values[index];
-        const int quantity = g->u.activity.values[index + 1];
-        const bool is_worn = position < -1;
-        const bool is_weapon = position == -1;
-        if( is_worn ) {
-            item &armor = g->u.i_at( position );
-            assert( !armor.is_null() );
-            selected_worn_items.push_back( &armor );
-            worn_item_quantities.push_back( quantity );
-        } else  if( is_weapon ) {
-            selected_items.push_back( &g->u.weapon );
-            item_quantities.push_back( quantity );
-        } else {
-            // We MUST have pointers to these items, and this is the only way I can see to get them.
-            std::list<item> &stack = ( std::list<item> & )g->u.inv.const_stack( position );
-            int items_dropped = 0;
-            for( auto &elem : stack ) {
-                selected_items.push_back( &elem );
-                if( elem.count_by_charges() ) {
-                    const int qty_to_drop = std::min( quantity - items_dropped, ( int )elem.charges );
-                    item_quantities.push_back( qty_to_drop );
-                    items_dropped += qty_to_drop;
-                } else {
-                    item_quantities.push_back( 1 );
-                    items_dropped++;
-                }
-                if( items_dropped >= quantity ) {
-                    break;
-                }
+    if( pet.inv.empty() ) {
```

Comment:
Yes, the `pet.inv` check is a sanity check. It should never really happen, since the caller checks this prior to calling the function and never calls it for an empty handed zed. But it does make sense to show a `debugmsg` here.